### ISSUE TITLE
Focus filename after creating file/ notebook through file browser

### ISSUE THREAD

**Issue by ivirshup** [jupyterlab/jupyterlab#9156]
### Problem

I'm always frustrated when I create any file, because it takes a few steps to set the filename.

I would like it if the filename was focussed after creation, since this would cause less friction, and match the UI for directory creation. It would probably also lead to me having fewer `untitled.{ext}` files around.

### Proposed Solution

Focus the filename after creating a new file or notebook – at least when they are created through the file browser.

### Additional context

This is the current behaviour for creating directories, but not files:

![output](https://user-images.githubusercontent.com/8238804/95827423-7501ec00-0d7f-11eb-8b0a-aea338e3d9a0.gif)

![output2](https://user-images.githubusercontent.com/8238804/95827922-11c48980-0d80-11eb-99b7-9393a67ca2a5.gif)

This would fit much more closely with other editors UI's, VSCode for example:

![output3](https://user-images.githubusercontent.com/8238804/95828430-b5159e80-0d80-11eb-85aa-8be62d951867.gif)

I think this would also solve #5492, since it would become trivial to create a file with whatever extension you want.

**Comment by philippeller** [comment]
Yes!! Everyone ends up with way too many files named `Untitled`, which just should not exist by default. It makes life harder rather than simpler in the long run. I would even go so far as to prompt the user to set a filename and not default it to `Untitled`

**Comment by mocquin** [comment]
+1, and I extend this behaviour for creation through "File->New->..."
